**Name: Maria Master**

**Class: INST 314 (0202)**

## Dataset 3: Movie Industry

**1. Describe how you found the data.  If you used a search engine, provide the exact search engine and search terms you used. Otherwise, describe how you came to find the data.**

**A:** I found this dataset in Kaggle after I looked up "Movies" in the search bar.

**2. Give a short verbal description of the data. Just a couple of sentences, like you were describing it briefly to someone in person, just to tell them generally what it is.**

**A:** This dataset is a collection of movies from 1986 - 2016. It contains more information about each of the movie such as the budget, IMDb score, rating, votes and more. This dataset can help provide insights into how the movie industry is doing within the timeframe of 1980s - 2016. Even though there is no information regarding streaming services, we can still potentially find trends of whether the movie industry is dying or not. 

**3. Do you have any information about how the data was collected?  If so, briefly describe the data collection process.  If not, take a guess how it might have been collected, and describe why you think that.**

**A:** The data was scraped from the HTML of IMDb using a Python Script. 

**4. a) Where is the data located (URL, etc.)?  This should be specific enough that the instructor/graders can find and download thedata themselves.If the instructor/graders cannot access or download the data themselves, describe what is preventing that (e.g., this is personal data on your own computer), and describe how you plan to share the data.**

**A:** URL: https://www.kaggle.com/datasets/danielgrijalvas/movies

If the data can not be accessed using the URL above, I will share the csv file with the instructional team. PLEASE make sure to upload the data in the same workspace as this Jupyter Notebook.

**4. b) File size (in KB, MB, or GB)**

**A:** 1.35 MB

**4. c) File format (extension, CSV, JSON, Excel, etc.)**

**A:** CSV

**5. How do you know you have permission to use and share the data?  For example, if there are any licenses, describe those, or if there is a URL to a page that describes permissions, you can link to that. Otherwise, describe your rationale for why you think you can analyze and share the data.**

**A:** The license is: CC0: Public Domain. This is a Creative Commons Zero license. Therefore, the author is waiving their right of copyright and the data can be accessed in the public domain and used for any purposes without restrictions.

**6. Do you think there is any kind of privacy or ethical risk by using/sharing the data?  Describe your rationale.**

**A:** I do not think there are any severe any privacy or ethical risk using/sharing this data. The data was scraped from IMDb website. Therefore, any user information was anonymized and not presented in the dataset. However, there could be movie titles that are sensitive to some.

**7. Load the data into your analysis program (R or Python).  Paste in the code you used that worked.**

In [11]:
# import pandas
import pandas as pd

# create dataframe
df = pd.read_csv("movies.csv", sep = ",")

**8. Use R/Python to look at the first few rows of data.  Paste in the code you used to do that.**

In [12]:
# print the top 5 records
print(df.head(5))

                                             name rating      genre  year  \
0                                     The Shining      R      Drama  1980   
1                                 The Blue Lagoon      R  Adventure  1980   
2  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
3                                       Airplane!     PG     Comedy  1980   
4                                      Caddyshack      R     Comedy  1980   

                        released  score      votes         director  \
0  June 13, 1980 (United States)    8.4   927000.0  Stanley Kubrick   
1   July 2, 1980 (United States)    5.8    65000.0   Randal Kleiser   
2  June 20, 1980 (United States)    8.7  1200000.0   Irvin Kershner   
3   July 2, 1980 (United States)    7.7   221000.0     Jim Abrahams   
4  July 25, 1980 (United States)    7.3   108000.0     Harold Ramis   

                    writer            star         country      budget  \
0             Stephen King  Jack Nic

**9. Are you surprised by anything in the first few rows?  What kind of values did you expect to see?  Your answer can be general; you don't have to describe every variable/column.**

**A:** I did not find anything extremely surprising in the first few rows of data as they all seem reasonable given their variable names. However, the fact that the movies date all the way back to 1980 is a little surprising as I was hoping for more recent movies. Lastly, columns with text data could be more difficult to analyze. 

**10. How many columns/variables of numeric data are there?  Note that your data *must* have at least
two columns of numeric data.  If not, keep looking!**

**A:** There are 6 numeric columns. 

**11.  Is there non-numeric data?  What kinds of non-numeric data do there appear to be?**

**A:** Yes, there are 8 non-numeric variables. One of them contains the release date along with the country of release. Even though I assumed it could be datetime datatype, it probably is not as also contains the country's name. The rest are also string data type (text).

**12. How many rows/records are there?  How many columns/variables?**

**A:** There are 7668 rows/records. There are 15 columns/variables. 

**13. Select a numeric column that appears to be interesting.  Describe why you think it might be interesting.**

**A:** An interesting numeric column is "score". In general, people check IMDb score for a movie to find out the quality of the movie as critics score them as well. Therefore, analyzing the score can help us gain insights into the user's preferences for different genres. 

**14. Make an educated guess about what you think the minimum and maximum values of this numeric variable are. Describe your reasoning.**

**A:** I think the lowest score would be 1 and the highest would be 9. It is very rare to see a perfect 10/10 score for a movie. 

**15. Get the actual minimum and maximum values, and paste in the code you used to get these values.**

In [13]:
# find the min and max value of the numeric variable "radius_worst"
min_val = df["score"].min()
max_val = df["score"].max()

# print the min and max value 
print("Minimum value: ", min_val)
print("Maximum value: ", max_val)

Minimum value:  1.9
Maximum value:  9.3


**16. Were you surprised by the actual values?  Describe what you think about why they are defying your expectations.**

**A:** I was not surprised by the maximum value but a little surprised by the minimum value. It is a surprising to think that the lowest score for all the thousands of movies from the 1980s to 2016 is a 2 and there are no movies worse than a 2. 

**17. Can you see any patterns of missing data in this data set? Describe what you see, and paste in code you used to look.**

**A:** There are missing values. Please check Option 2 to see how many values are missing for each column/variable. 

In [14]:
""" 
Option 1: 
- Use .isna() pandas function to check whether there is a missing value or not.
- Will return true if yes, and False if no. """

print(df.isna().head(5))

    name  rating  genre   year  released  score  votes  director  writer  \
0  False   False  False  False     False  False  False     False   False   
1  False   False  False  False     False  False  False     False   False   
2  False   False  False  False     False  False  False     False   False   
3  False   False  False  False     False  False  False     False   False   
4  False   False  False  False     False  False  False     False   False   

    star  country  budget  gross  company  runtime  
0  False    False   False  False    False    False  
1  False    False   False  False    False    False  
2  False    False   False  False    False    False  
3  False    False   False  False    False    False  
4  False    False   False  False    False    False  


In [15]:
""" 
Option 2:
- Use the .isna() function with .sum() function. This will count any instances of True that 
the .isna() function will return. If the sum is 0, there are no missing record instances."""

print(df.isna().sum())

name           0
rating        77
genre          0
year           0
released       2
score          3
votes          3
director       0
writer         3
star           1
country        3
budget      2171
gross        189
company       17
runtime        4
dtype: int64


**18. Describe anything else you found surprising or noteworthy about the data.**

**A:** It is important to note that the budget column is missing a lot of values (2171). There are a lot of outliers when it comes to movie runtimes as well. The rating column could potentially have more options other than R and PG. Lastly, once again, this data comes directly from the users. Therefore, the data is subjective (such as votes, score).

**19. Is this data interesting to you?  Why or why not?  What kinds of questions do you think it might be able to answer?**

**A:** This is an interesting dataset. It is a large dataset and contains valuable information about movies in the age of streaming. Here are some of the questions. that can be answered using this dataset:

- What is the most preferred genre in the movie industry?
- Is there a relationship between the company and the score of the movie?
- Are there any trends in budget and gross?
- What are some of the leading factors that contribute to the movie's success?

These are only some of the questions. There could possibly be many more.